unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -145,48 +139,9 @@ import Unison.Util.TransitiveClosure (transitiveClosure)
 import Unison.Var (Var)
 import qualified Unison.Var as Var
 import qualified Unison.WatchKind as WK
-
-type F m i v = Free (Command m i v)
-
--- type (Action m i v) a
-type Action m i v = MaybeT (StateT (LoopState m v) (F m i v))
```

Comment:
In order to split up HandleInput, `Action` needs to be in a separate module so that individual input-handling modules can depend on it.

I moved all of the state related to the Action Monad to `HandleInput.Action`